src/main/java/ch/njol/skript/expressions/ExprNearestEntity.java

```diff
@@ -0,0 +1,127 @@
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import ch.njol.skript.lang.util.SimpleExpression;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+@Name("Nearest Entity")
+@Description("Gets the entity nearest to a location or another entity.")
+@Examples({"kill the nearest pig and cow relative to player",
+	"teleport player to the nearest cow relative to player",
+	"teleport player to the nearest entity relative to player",
+	"on click:\n\tkill nearest pig"})
+@Since("INSERT VERSION")
+public class ExprNearestEntity extends SimpleExpression<Entity> {
+
+	static {
+		Skript.registerExpression(ExprNearestEntity.class, Entity.class, ExpressionType.SIMPLE,
+			"[the] nearest %*entitydatas% [[relative] to %entity/location%]",
+			"[the] %*entitydatas% nearest [to %entity/location%]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	@NonNull
+	private EntityData<?>[] entityDatas;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	@NonNull
+	private Expression<?> relativeTo;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		entityDatas = ((Literal<EntityData<?>>) exprs[0]).getArray();
+		if (entityDatas.length != Arrays.stream(entityDatas).distinct().count()) {
+			Skript.error("Entity list may not contain duplicate entities");
+			return false;
+		}
+		relativeTo = exprs[1];
+		return true;
+	}
+
+
+	@Nullable
+	private Entity getNearestEntity(EntityData<?> entityData, Location relativePoint, @Nullable Entity excludedEntity) {
+		Entity nearestEntity = null;
+		double nearestDistance = -1;
+		for (Entity entity : relativePoint.getWorld().getEntitiesByClass(entityData.getType())) {
+			if (entity != excludedEntity && entityData.isInstance(entity)) {
+				double distance = entity.getLocation().distance(relativePoint);
+				if (nearestEntity == null || distance < nearestDistance) {
+					nearestDistance = distance;
+					nearestEntity = entity;
+				}
+			}
+		}
+		return nearestEntity;
+	}
+
+	@Override
+	protected Entity[] get(Event e) {
+		Object relativeTo = this.relativeTo.getSingle(e);
+		if (relativeTo == null) {
+			return new Entity[0];
```

Comment:
Rather than returning an empty entity array here, we can just return null.

Reply:
in this case, i'm returning a 0 length array instead of null because it avoids `SimpleExpression#getArray` running extra null checks:

https://github.com/SkriptLang/Skript/blob/4672477b407e421340431ca7a7330176c59e9134/src/main/java/ch/njol/skript/lang/util/SimpleExpression.java#L98-L139